.github/workflows/ci-example-notebooks.yml

```diff
@@ -0,0 +1,80 @@
+# This is a basic workflow to help you get started with Actions
+
+name: CI Example Notebooks
+
+# Controls when the action will run.
+on:
+  # Triggers the workflow on push or pull request events but only for the master branch
+  push:
+    branches:
+      - master
+      - release*
+  pull_request:
+    branches:
+      - master
+      - release*
+
+  # Allows you to run this workflow manually from the Actions tab
+  workflow_dispatch:
+
+defaults:
+  run:
+    shell: bash
+
+jobs:
+  unittests:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        notebook-file: [
+            "DRISE.ipynb",
+            "MNIST_scikit_saliency.ipynb",
+            "OcclusionSaliency.ipynb",
+            "SimilarityScoring.ipynb",
+            "SuperPixelSaliency.ipynb",
+            "VIAME_OcclusionSaliency.ipynb",
```

Comment:
Just to check my understanding- is this workflow run on every PR? Do you anticipate that we will hit any data usage limits if the number of covered notebooks grows and we have to run through every notebook?

Reply:
> Just to check my understanding- is this workflow run on every PR?

Yes, as well as every modification of the PR.

> Do you anticipate that we will hit any data usage limits if the number of covered notebooks grows and we have to run through every notebook?

I don't currently know what aspects of our notebooks will be applicable to a data-cap. Some aspects like downloading an image from flikr or torch are relatively small and infrequent. OpenML dataset might be an exception. If we hit slowdowns or data-limits we can enable more aggressive caching on the action side to drastically bring down the frequency of download requests/requirements.

I also set up this list to be explicit for now to give us an axis of control if such a described issue comes to the forefront to not block *all* notebook CI if one or two become problem children.

I'm not opposed to an alternative recommendation if someone comes in with one.

Reply:
Sounds good, I don't have any better ideas, so we can go with this for now. One thing to note is that the VIAME notebook downloads a 5 GB model file (from viame.kitware.com), which is something that we might want to revisit.